common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -724,25 +727,20 @@ def login_analytics(strategy, auth_entry, current_partial=None, *args, **kwargs)
 
 
 @partial.partial
-def associate_by_email_if_login_api(auth_entry, backend, details, user, current_partial=None, *args, **kwargs):  # lint-amnesty, pylint: disable=keyword-arg-before-vararg
+def associate_by_email_if_third_party_auth_is_required(auth_entry, backend, details, user, strategy, *args, **kwargs):
```

Comment:
Also, Please make this code specific to OAuth functionality. As currently it is generic and will impact the already existing SAML flow for this part. Please take a look at the following link.
cc: @waheedahmed @moconnell1453 
https://github.com/edx/edx-platform/blob/0ca2295e8f8a953790cca8f0fe09787f2be9c408/common/djangoapps/third_party_auth/pipeline.py#L756